samples/bluetooth/mesh/light_dimmer/src/model_handler.c

```diff
@@ -0,0 +1,308 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+/**
+ * @file
+ * @brief Model handler for the Light Dimmer Controller.
+ *
+ */
+#include <zephyr/bluetooth/bluetooth.h>
+#include <zephyr/bluetooth/mesh/proxy.h>
+#include <bluetooth/mesh/models.h>
+#include <dk_buttons_and_leds.h>
+#include "model_handler.h"
+#include <time.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(dimmer_sample, LOG_LEVEL_INF);
+
+#define SHORT_PRESS_THRESHOLD_MS 300
+#define LONG_PRESS_TIMEOUT_SEC K_SECONDS(20)
+#define DELTA_MOVE_STEP 1000
+
+enum button_state_t { BUTTON_STATE_RELEASED, BUTTON_STATE_PRESSED };
```

Comment:
The `_t` suffix is reserved by POSIX for library types, and should not be used for internal type names.